src/network/core/address.cpp

```diff
@@ -242,6 +242,11 @@ SOCKET NetworkAddress::Resolve(int family, int socktype, int flags, SocketList *
 		strecpy(this->hostname, fam == AF_INET ? "0.0.0.0" : "::", lastof(this->hostname));
 	}
 
+#ifdef __EMSCRIPTEN__
```

Comment:
would appreciate comment about why this is necessary

Reply:
emscripten getaddrinfo fails when given AI_ADDRCONFIG.
reset_hostname is set because comparing addresses doesn't match entries properly, thus workaround is to compare strings returned from getnameinfo